src/compile/mark/geoshape.ts

```diff
@@ -0,0 +1,37 @@
+import {UnitModel} from '../unit';
+import * as mixins from './mixins';
+
+import {Config} from '../../config';
+import {Field, field, FieldDef, isFieldDef} from '../../fielddef';
+import {GEOSHAPE} from '../../mark';
+import {GEOJSON} from '../../type';
+import {VgGeoShapeTransform, VgPostEncodingTransform} from '../../vega.schema';
+import {getMarkConfig} from '../common';
+import {MarkCompiler} from './base';
+
+export const geoshape: MarkCompiler = {
+  vgMark: 'shape',
+  encodeEntry: (model: UnitModel) => {
+    const {config} = model;
+    return {
+      ...mixins.color(model),
+      ...mixins.text(model, 'tooltip'),
+      ...mixins.nonPosition('opacity', model)
```

Comment:
If we merge the href PR before we merge this one, we will have to add href here too :) 

https://github.com/vega/vega-lite/pull/3229

Reply:
FYI, to make this easier for the future -- I added https://github.com/vega/vega-lite/pull/3280#event-1433598724